### PR TITLE
Add architecture docs for snapshot stack

### PR DESCRIPTION
**Pull request by kenoir**
Also adds a missing requirements.txt ... oops.

Easy view of [`README.md`](https://github.com/wellcomecollection/catalogue/blob/9ade6453235bf900b11c1ae4873a7eba26b4cd17/snapshots/README.md)

Completes: https://github.com/wellcomecollection/platform/issues/4817

### CODE REVIEW

File: snapshots/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Contains:
+
+- snapshot_generator: inspect the production Catalogue Elasticsearch index and produce a snapshot of works using their display model.
+- snapshot_scheduler: a lambda to schedule snapshots (triggers the snapshot generator). Triggered daily by CloudWatch events.
+- snapshot_recorder: notified by SNS -> SQS from the snapshot_generator about completed snapshots, recording metadata in a reporting cluster Elasticsearch index.
+- snapshot_reporter: reports on daily snapshots in team Slack, will provide notification on failure. Triggered daily by CloudWatch events.
+
+## Architecture
+
+![](architecture.png)
```
Comment (jamesgorrie) [inline snapshots/README.md:16]:
```suggestion
![Architecture diagram for catalogue snapshots](architecture.png)
```

File: snapshots/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Deployment
+
+### Steps
+
+The snapshot_generator is deployed alongside the Catalogue API (as specified in `.wellcome_project`), in order that it remains in sync with the current display model. You will need to use the `weco-deploy` tool.
```
Comment (jamesgorrie) [inline snapshots/README.md:22]:
```suggestion
The snapshot_generator is deployed alongside the Catalogue API (as specified in `.wellcome_project`), in order that it remains in sync with the current display model. You will need to use the [`weco-deploy`](https://github.com/wellcomecollection/weco-deploy) tool.
```

File: snapshots/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+```
+weco-deploy --project-id catalogue_api deploy --environment-id prod
+```
+
+The snapshot_recorder, snapshot_scheduler & snapshot_reporter lambdas should be published as required using the relevant `make` commands. A terraform apply is then required to update those lambdas in AWS.
+
+For example:
```
Comment (jamesgorrie) [inline snapshots/README.md:30]: This doesn't happen in CI?

File: snapshots/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Services for creating, recording and reporting on Catalogue API snapshots.
+
+## Overview
+
+Contains:
+
+- snapshot_generator: inspect the production Catalogue Elasticsearch index and produce a snapshot of works using their display model.
+- snapshot_scheduler: a lambda to schedule snapshots (triggers the snapshot generator). Triggered daily by CloudWatch events.
+- snapshot_recorder: notified by SNS -> SQS from the snapshot_generator about completed snapshots, recording metadata in a reporting cluster Elasticsearch index.
```
Comment (alexwlchan) [inline snapshots/README.md:11]: It's just SNS, no SQS involved.

File: snapshots/README.md (hunk 3 above)
Comment (alexwlchan) [reply snapshots/README.md:30]: Not currently, we took the Lambdas out of CI a while back to free up Travis workers.

**Review by alexwlchan — APPROVED** [review]

File: snapshots/README.md (hunk 4 above)
Comment (kenoir) [reply snapshots/README.md:11]: I knew that!

**Pull request merged**